utils/azureCredentialsManager.ts

```diff
@@ -0,0 +1,140 @@
+import { SubscriptionClient, ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { AzureAccount } from '../typings/azure-account.api';
+import { ServiceClientCredentials } from 'ms-rest';
+import { AsyncPool } from '../utils/asyncpool';
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import * as ContainerModels from '../node_modules/azure-arm-containerregistry/lib/models';
+import { ResourceGroup, ResourceGroupListResult } from "azure-arm-resource/lib/resource/models";
+import { MAX_CONCURRENT_SUBSCRIPTON_REQUESTS } from './constants';
+
+/* Singleton for facilitating communication with Azure account services by providing extended shared
+  functionality and extension wide access to azureAccount. Tool for internal use.
+  Authors: Esteban Rey L, Jackson Stokes
+*/
+
+export class AzureCredentialsManager {
+
+    //SETUP
+    private static _instance: AzureCredentialsManager = new AzureCredentialsManager();
+    private azureAccount: AzureAccount;
+
+    private constructor() {
+        AzureCredentialsManager._instance = this;
```

Comment:
You can remove this: unreachable code.

Reply:
Would I just maintain an empty private constructor then?

Reply:
Yes, making the constructor private ensures this class cannot be instantiated. Hence, the only way is to use static object creation `getInstance` where a singleton instance is created only the first time its called. Singleton is just a fancy way of using a global shared object.